core/src/repair_weight.rs

```diff
@@ -249,28 +271,54 @@ impl RepairWeight {
     }
 
     /// Split `slot` and descendants into an orphan tree in repair weighting
-    /// These orphaned slots should be removed from `unrooted_slots` as on proper repair these slots might
-    /// now be part of the rooted path
+    /// Even if these orphaned slots were previously pruned, they should be added back properly as
+    /// on repair they might now be part of the rooted path.
+    /// Assumes that `slot` is greater than `self.root`
```

Comment:
Some more detail here would be helpful

1.  What does `added back properly` mean?
2. `On repair they might now be part of the rooted path.` Details of this mechanism would be good for documentation